src/SourceGenerators/Uno.UI.SourceGenerators.Internal/AvoidMultipleGettersAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Uno.UI.SourceGenerators.Internal;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class AvoidMultipleGettersAnalyzer : DiagnosticAnalyzer
+{
+	private static DiagnosticDescriptor _descriptor = new("UnoMultipleGetValue", "Avoid duplicate DP access", "Avoid duplicate DP access", "Performance", DiagnosticSeverity.Error, isEnabledByDefault: true);
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(_descriptor);
+
+	public override void Initialize(AnalysisContext context)
+	{
+		context.EnableConcurrentExecution();
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+		context.RegisterOperationBlockAction(context =>
+		{
+			foreach (var operationBlock in context.OperationBlocks)
+			{
+				var cfg = context.GetControlFlowGraph(operationBlock);
+				var entryBlock = cfg.Blocks[0];
+				// This is not a good implementation. It only handle the case where the duplicate access is in the same BasicBlock.
+				// There are more cases that are currently not handled.
```

Comment:
@jeromelaban Should I try to look into improving this? I'm not very familiar with DFA but can give a try if needed.

Reply:
I'm not familiar with this either, though if there's a performance problem with the analyzer, we may want to run it only in one of the platforms in CI.